### PR TITLE
test: add e2e smoke tests for headless-ssr utils

### PR DESCRIPTION
**Pull request by mrrajamanickam-coveo**
* Add e2e cypress smoke tests using the headless-ssr sample page
   - Add a corresponding new GH action to run these tests
* Modify sample page to include a timestamp 
   - This is then parsed by the tests to verify the page is rendered in both SSR and CSR
image

## Check before merging

- [ ] Cypress v12 is being used
- [ ] ~~Next CI cache is being used~~ Next Cache needs to be [moved to root for GH cache action to work](https://github.com/actions/setup-python/issues/476). Can revisit when next build times are significant.

https://coveord.atlassian.net/browse/KIT-2643

### CODE REVIEW

File: packages/samples/headless-ssr/cypress/e2e/ssr.cy.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    cy.get(msgSelector).should('have.text', numResultsMsg);
+    cy.get('li').should('have.length', numResults);
+  });
+
+  it('renders result list in SSR and then in CSR', () => {
+    const interceptAlias = 'searchResults';
+    cy.intercept('/').as(interceptAlias);
+
+    cy.visit('/');
+    cy.wait(`@${interceptAlias}`).then((intercept) => {
```
Comment (louis-bompart) [inline packages/samples/headless-ssr/cypress/e2e/ssr.cy.ts:32]: You should be able to use an async/await in the `it` function.

File: packages/samples/headless-ssr/cypress/support/commands.ts
Comment (louis-bompart) [inline packages/samples/headless-ssr/cypress/support/commands.ts:1]: errr. what is it good for?

**Review by louis-bompart — APPROVED** [review]
Look nice!
Wondering how we should proceed regarding Cypress:
- use the same version as the other packages but then have another package to update when we do the update batch
- use the latest version and saves one update, but be out-of-sync with the other package, causing us to install cypress twice

File: packages/samples/headless-ssr/cypress/support/commands.ts
Comment (olamothe) [reply packages/samples/headless-ssr/cypress/support/commands.ts:1]: IIRC those are auto generated by Cypress when you initialize a new project.

We can delete those indeed.

File: packages/samples/headless-ssr/cypress/e2e/ssr.cy.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      );
+      expect(ssrTimestamp).to.not.be.undefined;
+      cy.get(timestampSelector).should((timeStampElement) => {
+        const csrTimestamp = Date.parse(timeStampElement.text());
+        expect(csrTimestamp).to.be.greaterThan(ssrTimestamp);
+      });
+    });
+  });
+
+  it('should pass the web-vitals audits', () => {
```
Comment (olamothe) [inline packages/samples/headless-ssr/cypress/e2e/ssr.cy.ts:48]: Is this something we want to really fail tests/PR ? 

Or simply be aware of/warned of ?

**Review by olamothe — APPROVED** [review]

File: packages/samples/headless-ssr/cypress/e2e/ssr.cy.ts (hunk 2 above)
Comment (btaillon) [reply packages/samples/headless-ssr/cypress/e2e/ssr.cy.ts:48]: I'm not too worried about having to adjust this over time.

**Review by btaillon — APPROVED** [review]

File: packages/samples/headless-ssr/cypress/e2e/ssr.cy.ts (hunk 1 above)
Comment (mrrajamanickam-coveo) [reply packages/samples/headless-ssr/cypress/e2e/ssr.cy.ts:32]: `cy.wait` returns a `Chainable<Interception>` and not a `Promise`. Using `await` results in the error: `Type of 'await' operand must either be a valid promise or must not contain a callable 'then' member.`

**Pull request merged**